crates/sovereign-sdk/adapters/risc0-bonsai/src/host.rs

```diff
@@ -494,19 +508,20 @@ impl<'a> ZkvmHost for Risc0BonsaiHost<'a> {
         }
     }
 
-    fn extract_output<Da: sov_rollup_interface::da::DaSpec, Root: Serialize + DeserializeOwned>(
+    fn extract_output<Da: sov_rollup_interface::da::DaSpec, Root: BorshDeserialize>(
         proof: &Proof,
     ) -> Result<sov_rollup_interface::zk::StateTransition<Da, Root>, Self::Error> {
-        match proof {
+        let journal = match proof {
             Proof::PublicInput(journal) => {
                 let journal: Journal = bincode::deserialize(journal)?;
-                Ok(journal.decode()?)
+                journal
             }
             Proof::Full(data) => {
                 let receipt: Receipt = bincode::deserialize(data)?;
-                Ok(receipt.journal.decode()?)
+                receipt.journal
             }
-        }
+        };
+        Ok(BorshDeserialize::try_from_slice(&journal.bytes)?)
```

Comment:
i thought we were going to use borsh also for Receipt encoding and decoding (for writing to da and reading from da)

Reply:
I havent found the place where they serialize it with bincode yet

Reply:
ok. this can also be done later. I am opening an issue.